spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/LabeledSecretContextToSourceDataProvider.java

```diff
@@ -77,10 +81,43 @@ public KubernetesClientContextToSourceData get() {
 						createLabelsSelector(labels), null, null, null, null, null).getItems();
 
 				if (!secrets.isEmpty()) {
-					sourceName = secrets.stream().map(V1Secret::getMetadata).map(V1ObjectMeta::getName)
+
+					for (V1Secret secret : secrets) {
```

Comment:
I have no idea how you do it, but you always ping places where there is a bug. this is awesome! I have realized that I have a small issue here: since the order is undefined, we still need a defined order for the prefix, otherwise users will not be able to wire the correct property all the time...

I've added `sorted` when computing the prefix where we find multiple secrets. For example, if based on label: `{color:blue}` we find two secrets `secret-a` and `secret-b`, prefix is going to be `secret-a.secret-b` (sorted names).

thank you for bringing this up.

Reply:
HAHAHA you are giving more credit than I deserve!

This is still probably worth adding to the docs IMO

Reply:
done.